lib/utils/classes.py

```diff
@@ -22,6 +22,20 @@ class Stock:
     annual_reports: List[Document]
     credit_ratings: List[Document]
 
+    financial_ratios: pd.DataFrame
+
+    roce_above_10: bool
+    rising_roce: bool
+    roce_avg_bps_increase: float
+
+    yoy_margin_increase_count: int
+    negative_margin_count: int
+    margin_score: int
+
+    controlled_dpr: bool
```

Comment:
I am a bit concerned about having so many properties on the class, that makes the class confusing - This looks like  there should be a substructure, kinda like this:

Having a structure like this would greatly increase readability and improve code 
```
roce:
    above_10: bool
    rising: bool
    avg_bps_increase: float

margin:
    yoy_increase_count: int
    negative_count: int
    score: float

dividend:
    ...
```